SU2_CFD/include/objectives/turbomachinery/CTurbomachineryPerformance.hpp

```diff
@@ -0,0 +1,254 @@
+/*!
+ * \file CTurbomachineryPerformance.hpp
+ * \brief Headers of the Turbomachinery Performance class.
+ * \author S. Vitale
+ * \version 7.0.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <algorithm>
+#include <iterator>
+#include <numeric>
+#include <boost/bind.hpp>
+#include <boost/range/algorithm_ext/for_each.hpp>
+#include "../../../../Common/include/mpi_structure.hpp"
+#include "../../../../Common/include/geometry/CGeometry.hpp"
+#include "../../../../Common/include/CConfig.hpp"
+#include "../../fluid_model.hpp"
+
+using namespace std;
+
+class CTurbomachineryPrimitiveState {
+  private:
+  su2double Density, Pressure, TangVelocity;
+  vector <su2double> Velocity;
+  public:
+  CTurbomachineryPrimitiveState();
+
+  CTurbomachineryPrimitiveState(su2double density, su2double pressure, su2double *velocity, unsigned short nDim,
+                                su2double tangVel);
+
+  ~CTurbomachineryPrimitiveState();
```

Comment:
If the default ctor and the dtor are empty I prefer not to declare/define them at all, if you want to use the `= default;` syntax.
Because user defined destructors disable the compiler-generated move operations.